### PR TITLE
Allow PagerPan to accept a transitionConfig allowing custom timing/animation configs

### LINKED ISSUE #173: How to configure tabs slide transition?
When I set new index to state of tabview (i.e. switch tabs programatically, from a button click) tabs switch too fast for me, I'd like to have a slower transition. Docs say there's `configureTransition` callback, but don't say how should it be written and how does that 'transition configuration' that it should return should look like. Could you please, pour some light on it and update the docs?

### LINKED ISSUE #430: configureTransition() is an unused property?
### Actual behaviour
`configureTransition()` prop doesn't affect the transition between the tabs. Even my IDE gives me a hint that the prop isn't used.

### Expected behaviour
I expect to modify the behavior of the tab transitions as described in the `README.md`.

### Code sample, screenshots
```
{
     ... other tab config
     configureTransition: (currentTransitionProps,nextTransitionProps) => ({
          timing: Animated.spring,
          tension: 1000,
          friction: 100,
     }),
     ... other tab config
}
```

### What have you tried
I have reviewed the implementation and figured out that it is going to be forwarded correctly to the `TabViewPagerPan` component. Since then there isn't any further usage of the function. The `transitionTo()` function, which is responsible for performing the transition (I suppose?) is based on the `DefaultTransitionSpec`.

`const { timing, ...transitionConfig } = DefaultTransitionSpec;`

Doesn't even work with the `react-navigation` sample: https://snack.expo.io/Sye_jJWSM

### PR DESCRIPTION
**Pull request by levibuzolic**
`configureTransition` prop was accidentally removed from `PagerPan` during the transition to 1.0 this PR adds a `transitionConfig` prop which serves the same purpose however without the complex function API that `configureTransition` originally had.

TL;DR adds `transitionConfig` prop allowing custom timing/springs for `PagerPan`.

Closes #430
Closes #173

/cc @satya164

### CODE REVIEW

**Comment by levibuzolic** [comment]
Anybody? @satya164 ? 

Quite keen to get this over the line, let me know if there's anything I can do to help.

**Comment by jdonald** [comment]
With this I was able to use a custom transition, but the format I'm used to with other react components is that `transitionConfig` is supposed to be a function that returns an object containing  `transitionSpec` e.g.:
```
const transitionConfig = () => ({
  transitionSpec: {
    duration: 0,
    timing: Animated.timing,
    easing: Easing.step0
  }
});
```
Other than `PropTypes.func`, something's amiss here that the code is assuming transitionConfig is transitionSpec.

**Comment by satya164** [comment]
Hey, I just released a new alpha `2.0.0-alpha.0` of the library. It's rewritten using [react-native-gesture-handler](https://github.com/kmagiera/react-native-gesture-handler) and [react-native-reanimated](https://github.com/kmagiera/react-native-reanimated) addresses a many platform specific bugs and performance problems. It's gone through a major refactor and there will be conflicts with your PR.

Please try the new version and see if it addresses your issue. If not, please open a new PR with the change.
